app/models/katello/glue/pulp/repo.rb

```diff
@@ -778,9 +771,11 @@ def _get_most_recent_sync_status
       end
     end
 
-    def full_path(smart_proxy = nil, force_https = false)
-      pulp_uri = URI.parse(smart_proxy ? smart_proxy.url : SETTINGS[:katello][:pulp][:url])
-      scheme   = (self.unprotected && !force_https) ? 'http' : 'https'
+    def full_path(smart_proxy, force_https = false)
+      capsule_content = ::Katello::CapsuleContent.new(smart_proxy)
+      pulp_uri = URI.parse(capsule_content.pulp_url)
+
+      scheme = (self.unprotected && !force_https) ? 'http' : 'https'
       if docker?
         "#{pulp_uri.host.downcase}:#{Setting['pulp_docker_registry_port']}/#{container_repository_name}"
```

Comment:
IMHO the proxy should also expose the port. That way you can have different registry ports per proxy. It also looks weird that this one doesn't have a scheme.

Reply:
Since the docker aspect works differently than just deriving where Pulp lives from the proxy I'd prefer not to touch this and expand the scope. I think this is a valid request that could be filed as a separate issue.